.circleci/config.yml

```diff
@@ -3,67 +3,174 @@
+        description: pip install -r requirements_test_all.txt
+        type: boolean
+        default: false
+    steps:
       - restore_cache:
           keys:
-            - v1-dependencies-{{ checksum "requirements_all.txt" }}-{{ checksum "requirements_test.txt" }}
-
+            - v1-<< parameters.python >>-<<# parameters.all >>{{ checksum "requirements_all.txt" }}<</ parameters.all>>-<<# parameters.test >>{{ checksum "requirements_test.txt" }}<</ parameters.test>>-<<# parameters.test_all >>{{ checksum "requirements_test_all.txt" }}<</ parameters.test_all>>
       - run:
           name: install dependencies
           command: |
             python3 -m venv venv
             . venv/bin/activate
-            pip install -q --progress-bar off -r requirements_all.txt -r requirements_test.txt -c homeassistant/package_constraints.txt
-
+            pip install -U pip
+            <<# parameters.all >>pip install -q --progress-bar off -r requirements_all.txt -c homeassistant/package_constraints.txt<</ parameters.all>>
+            <<# parameters.test >>pip install -q --progress-bar off -r requirements_test.txt -c homeassistant/package_constraints.txt<</ parameters.test>>
+            <<# parameters.test_all >>pip install -q --progress-bar off -r requirements_test_all.txt -c homeassistant/package_constraints.txt<</ parameters.test_all>>
       - save_cache:
           paths:
             - ./venv
-          key: v1-dependencies-{{ checksum "requirements_all.txt" }}-{{ checksum "requirements_test.txt" }}
+          key: v1-<< parameters.python >>-<<# parameters.all >>{{ checksum "requirements_all.txt" }}<</ parameters.all>>-<<# parameters.test >>{{ checksum "requirements_test.txt" }}<</ parameters.test>>-<<# parameters.test_all >>{{ checksum "requirements_test_all.txt" }}<</ parameters.test_all>>
 
+  install:
+    description: Install Home Assistant
+    steps:
       - run:
           name: install
           command: |
             . venv/bin/activate
             pip install --progress-bar off -e .
 
+jobs:
+
+  static-check:
+    executor:
+      name: python
+      tag: 3.7-stretch
+
+    steps:
+      - checkout
+      - docker-prereqs
+
       - run:
-          name: run lint
+          name: run static check
           command: |
+            python3 -m venv venv
             . venv/bin/activate
-            python script/gen_requirements_all.py validate
+            pip install -U pip
+            pip install --progress-bar off flake8
             flake8
-            pylint homeassistant
+
+  install-all-and-check-requirement:
+    executor:
+      name: python
+      tag: 3.7-stretch
+
+    steps:
+      - checkout
+      - docker-prereqs
+      - install-requirments:
+          python: 3.7-stretch
+          all: true
+          test: true
+      - install
+
+      - run:
+          name: run gen_requirements_all
+          command: |
+            . venv/bin/activate
+            python script/gen_requirements_all.py validate
```

Comment:
Validation should be done on a container that only has Home Assistant installed. That way we will catch imports that are outside of functions.

Reply:
Okay, then I move this to static check job